src/AddIns/Uno.UI.MediaPlayer.Skia.Gtk/GTKMediaPlayer.cs

```diff
@@ -212,79 +216,34 @@ private void UpdateVideoStretch()
 			}
 		});
 	}
+	protected void UpdateTransportControlHeight()
+	{
+		UIElement? container = VisualTreeHelper.GetParent(_videoContainer) as UIElement;
+		ContentPresenter? ContentPresenter = VisualTreeHelper.GetParent(container) as ContentPresenter;
+		ContentControl? ContentControl = VisualTreeHelper.GetParent(ContentPresenter) as ContentControl;
+		MediaPlayerPresenter? MediaPlayerPresenter = VisualTreeHelper.GetParent(ContentControl) as MediaPlayerPresenter;
+		UIElement? MediaPlayerPresenter2 = VisualTreeHelper.GetParent(MediaPlayerPresenter) as UIElement;
+
+		if (MediaPlayerPresenter2 != null)
+		{
+			foreach (var child in MediaPlayerPresenter2.GetChildren())
+			{
+				if (child is ContentPresenter)
+				{
+					_transportControlAdjust = (int)child.DesiredSize.Height;
+				}
+			}
+		}
+	}
 
 	protected override Size ArrangeOverride(Size finalSize)
 	{
-		var result = base.ArrangeOverride(finalSize);
 		UpdateVideoStretch();
+		var result = base.ArrangeOverride(finalSize);
```

Comment:
This feels the wrong order, and UpdateVideoStretch is async (it invokes dispatcher) so we shouldn't be invoking before we call the base ArrangeOverride.